festim/generic_simulation.py

```diff
@@ -418,6 +418,14 @@ def initialise(self):
                         self.dt.milestones.append(time)
                 self.dt.milestones.sort()
 
+        # set Soret to True
+        for export in self.exports:
+            if isinstance(export, festim.DerivedQuantities):
+                for q in export:
+                    if isinstance(q, festim.SurfaceFlux):
+                        q.soret = self.settings.soret
+                        q.T = self.T.T
+
```

Comment:
Do you want to set the soret flag independently? There is the same `for export in self.exports` loop above.

Reply:
Good shout! This would require a serious refactoring anyway

Reply:
Indeed. In fact, there are two loops that go through exports. I don't think they are needed, but some refactoring can be done elsewhere.